src/pages/ScoreReport.vue

```diff
@@ -883,150 +911,197 @@ const viewOptions = ref([
   { label: 'Raw Score', value: 'raw' },
 ]);
 
-const exportSelected = (selectedRows) => {
-  const computedExportData = _map(selectedRows, ({ user, scores }) => {
+/**
+ * Creates and formats the data for exporting user, score, and optionally, progress information to a CSV file.
+ *
+ * @NOTE This function generates a structured dataset based on user and score data, with optional inclusion of progress
+ * data. It ensures that the data is organized appropriately for export, including task-specific formatting and reliability checks.
+ * If progress data is included, it appends relevant progress information per task.
+ *
+ * @param {Object[]} rows - The array of user data and associated scores.
+ * @param {Object} rows[].user - The user object containing user details such as username, email, first name, and last name.
+ * @param {Object} rows[].scores - The scores object containing task-related score data for the user. It supports different
+ * score types (percent correct, raw scores, standard scores, etc.) based on task configuration.
+ * @param {boolean} [includeProgress=false] - Flag indicating whether to include task progress data in the export. If true,
+ * progress data will be fetched and appended for each task per user.
+ *
+ * @returns {Object[]} - The formatted data array, where each object represents a user and their associated scores.
+ * This data is ready for CSV export and optionally includes progress information.
+ *
+ * @NOTE This function also checks for the user's role (e.g., super admin) to determine additional fields (such as PID),
+ * handles task-specific score presentation based on configuration, and validates task reliability using engagement flags.
+ * If scores are found unreliable, the reliability reason is included. If the task is incomplete, it is marked as such.
+ */
+
+const createExportData = ({ rows, includeProgress = false }) => {
+  const computedExportData = _map(rows, ({ user, scores }) => {
     let tableRow = {
       Username: _get(user, 'username'),
+      Email: _get(user, 'email'), // This will only be used when exporting all rows
       First: _get(user, 'firstName'),
       Last: _get(user, 'lastName'),
       Grade: _get(user, 'grade'),
```

Comment:
No need for `lodash`, we can achieve the exact same using optional chaining:  

```suggestion
      Username: user?.username,
      Email: user?.email, // This will only be used when exporting all rows
      First: user?.firstName,
      Last: user?.lastName,
      Grade: user?.grade,
```